File: stepik-union/src/main/java/org/stepik/plugin/utils/ReformatUtils.java
package org.stepik.plugin.utils;

import com.intellij.codeInsight.actions.OptimizeImportsProcessor;
import com.intellij.codeInsight.actions.ReformatCodeProcessor;
import com.intellij.openapi.editor.Document;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiDocumentManager;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;

public class ReformatUtils {

    public static void reformatSelectedEditor(@NotNull Project project, @NotNull Document document) {
        PsiDocumentManager.getInstance(project).commitAllDocuments();

        PsiFile file = PsiDocumentManager.getInstance(project).getPsiFile(document);
        if (file == null)
            return;

        new ReformatCodeProcessor(new OptimizeImportsProcessor(project, file), false).run();
    }
}

File: stepik-core/src/main/kotlin/org/stepik/core/utils/ReformatUtils.kt
package org.stepik.core.utils

import com.intellij.codeInsight.actions.OptimizeImportsProcessor
import com.intellij.codeInsight.actions.ReformatCodeProcessor
import com.intellij.openapi.editor.Document
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiDocumentManager


fun Project.reformatDocument(document: Document) {
    PsiDocumentManager.getInstance(this).commitAllDocuments()

    val file = PsiDocumentManager.getInstance(this).getPsiFile(document) ?: return

    ReformatCodeProcessor(OptimizeImportsProcessor(this, file), false).run()
}
